### PR TITLE
daemon: Delete deprecated code to remove older cilium-envoy.log at startup

### PR DESCRIPTION
**Pull request by manalibhutiyani**
We had added a one time `cilium-envoy.log` removal, pre-1.0 stage to remove the cilium log in
older location on user machines running cilium-envoy setup.
This code simply removes this deprecated code to do this one-time cleanup since we are post
1.1 at this point, and have given enough time for this code to bake.

Fixes: #3050
Signed-Off-By: Manali Bhutiyani <manali@covalent.io>

```release-note
Don't remove old (pre-1.0) cilium-envoy.log on startup
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4518)

### CODE REVIEW

**Comment by manalibhutiyani** [comment]
test-me-please

**Comment by manalibhutiyani** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]
Minor nit, I think it'd be easier to read the release note for a user perspective if it was phrased more like this:

```
Don't remove old (pre-1.0) cilium-envoy.log on startup
```

Though that said I'm wondering if we really need to notify users about this at all.

Otherwise LGTM. I think that master is still feature-frozen for 1.1 so let's hold off on merging this?

**Comment by manalibhutiyani** [comment]
@joestringer :Changed the release note. As per your other comment, I feel this change it pretty minor and non-distruptive to qualify as a feature. IMHO I think merging it should be ok. I have changed the release - note to reflect that this is a non-feature.

**Comment by manalibhutiyani** [comment]
test-me-please

**Comment by manalibhutiyani** [comment]
test-me-please

**Comment by manalibhutiyani** [comment]
test-me-please

**Comment by aanm** [comment]
@manalibhutiyani any reason for the milestone removal?

**Comment by manalibhutiyani** [comment]
@aanm : Since this is a cleanup, I was planning to merge in master today, hence removed the milestone.

**Comment by manalibhutiyani** [comment]
@aanm :Can I get an approval from `agents` side if things look ok?

**Review by aanm — APPROVED** [review]
@manalibhutiyani let's keep it for 1.2 just to make sure users coming from pre-1.0 will have the file clean up

**Review by tgraf — APPROVED** [review]

**Pull request merged**
